### PR TITLE
Standardize TargetPlatformVersion format

### PR DESCRIPTION
**Pull request by sfoslund**

### CODE REVIEW

**Comment by sfoslund** [comment]
@wli3 @dsplaisted this fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1160292. I'm targeting preview 8 right now because we don't have these changes in master yet, but I can retarget once I merge preview 8 into master tomorrow.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs [hunk 1]
```diff
@@ -35,6 +35,25 @@ public void It_fails_if_windows_target_platform_version_is_invalid()
...
+        [WindowsOnlyFact]
+        public void It_succeeds_if_windows_target_platform_version_has_trailing_zeros()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "ValidWindowsVersion",
+                IsSdkProject = true,
+                TargetFrameworks = "net5.0"
+            };
+            testProject.AdditionalProperties["TargetPlatformIdentifier"] = "windows";
```
Comment (Nirmal4G) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs:47]:
```suggestion
            testProject.AdditionalProperties["TargetPlatformIdentifier"] = "Windows"; 
```

I thought TPI/TFI should be in PascalCasing, no?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets [hunk 2]
```diff
@@ -198,13 +198,22 @@ Copyright (c) .NET Foundation. All rights reserved.
...
 
-  <Target Name="_CheckForInvalidTargetPlatformVersion"
-      BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
-      Condition="'$(TargetPlatformVersion)' != '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">
+  <Target Name="_StandardizeTargetPlatformVersion"
+    BeforeTargets="ProcessFrameworkReferences"
+    Condition="'$(TargetPlatformVersion)' != '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">
     <ItemGroup>
-      <ValidTargetPlatformVersion Include="@(SupportedTargetPlatform)" Condition="'@(SupportedTargetPlatform)' != '' and '%(Identity)' == '$(TargetPlatformVersion)'" />
+      <ValidTargetPlatformVersion Include="@(SupportedTargetPlatform)" Condition="'@(SupportedTargetPlatform)' != '' and $([MSBuild]::VersionEquals(%(Identity), $(TargetPlatformVersion)))" />
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:205]: I'd suggest prefixing this with an underscore (ie `_ValidTargetPlatformVersion`) to indicate that it's an internal / temporary property.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs (hunk 1 above)
Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs:47]: We do normalize the casing to `Windows`.  That said, if a test isn't meant to be testing case normalization, it should probably use `Windows` if the platform is specified via the `TargetPlatformIdentifier` property, and `-windows` if it's specified in the `TargetFramework`, as those are the standard casings in each case.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs [hunk 3]
```diff
@@ -35,6 +35,25 @@ public void It_fails_if_windows_target_platform_version_is_invalid()
...
+                TargetFrameworks = "net5.0"
+            };
+            testProject.AdditionalProperties["TargetPlatformIdentifier"] = "windows"; 
+            testProject.AdditionalProperties["TargetPlatformVersion"] = "10.0.18362.0"; // We must set this manually because if we set it in the TFM we remove the trailing zeroes. 
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Pass();
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs:54]: I would also add to this test a check that the `TargetPlatformVersion` gets the trailing `.0` trimmed off.

**Review by dsplaisted — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets (hunk 2 above)
Comment (wli3) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:205]: nit: usually it is called "normalize"

**Review by wli3 — APPROVED** [review]

**Pull request merged**